jenkinsfiles/ubuntu.Jenkinsfile

```diff
@@ -0,0 +1,113 @@
+// Parameters:
+// - ENVIRONMENT
+// - VERSION
+// - GENESIS_PATH
+// - UBUNTU_VERSION
+// - GHC_VERSION
+// - FORCE_REBUILD
+
+@Library('concordium-pipelines') _
+
+Map rpc_port = [
+    mainnet: "10000",
+    testnet: "10001",
+    stagenet: "10500"
+]
+
+Map listen_port = [
+    mainnet: "8888",
+    testnet: "8889",
+    stagenet: "9500"
+]
+
+pipeline {
+    // Use jenkins-worker, as it has the keys for pushing to AWS.
+    agent { label 'jenkins-worker' }
+    environment {
+        // Extract version from code
+        CODE_VERSION = """${sh(
+                returnStdout: true,
+                script: '''awk '/version = / { print substr($3, 2, length($3)-2); exit }' concordium-node/Cargo.toml'''
+            )}""".trim()
+        // Use code version if the version param has not been set
+        OUT_VERSION = """${sh(
+                returnStdout: true,
+                script: "[[ -z '${VERSION}' ]] && echo '${CODE_VERSION}' || echo '${VERSION}'"
+            )}""".trim()
+        DOMAIN = concordiumDomain(ENVIRONMENT)
+        BUILD_FILE = "concordium-${ENVIRONMENT}-node_${CODE_VERSION}_amd64.deb"
+        OUTFILE = "s3://distribution.${DOMAIN}/deb/concordium-${ENVIRONMENT}-node_${OUT_VERSION}_amd64.deb"
+        GENESIS_HASH_PATH = "genesis/${GENESIS_PATH}/genesis_hash"
+        GENESIS_DAT_FILE = "genesis/${GENESIS_PATH}/genesis.dat"
+        ENVIRONMENT_CAP = environment.capitalize()
+        DATA_DIR = "./scripts/distribution/ubuntu-packages/template/data/"
+        RPC_PORT = "${rpc_port[environment]}"
+        LISTEN_PORT = "${listen_port[environment]}"
+    }
+    stages {
+        stage('Build static-node-binaries') {
+            when {
+                anyOf {
+                    // Build it if forced_rebuild is checked
+                    expression {
+                        return params.FORCE_REBUILD
+                    }
+
+                    // OR
+                    // Check if static-node-binaries image is present, if not build it.
+                    equals expected: "1",
+                    actual: "${sh script:'docker inspect --type=image static-node-binaries > /dev/null 2> /dev/null', returnStatus:true}"
+                }
```

Comment:
Seeing that this is a build script, will it ever make sense not to rebuild the static libraries? 

In the shell script that I have in the repository I added this because it makes it faster to build stagenet/testnet/mainnet versions at the same time, since it only has to build the static libraries once. However it is a bit tricky to use correctly since you have to remember to use the right version of the image :)

My guess would be that here we always want to rebuild, so I would simplify this and remove the "FORCE_REBUILD". Alternatively if reuse is useful, I would tag the image with the commit hash, not just `static-node-binaries`.

Reply:
Yeah, it has very limited usage, as the build agent, and any docker images on it, is killed after 10 minutes of inactivity. So I'll remove it to avoid any potential confusion.

Reply:
Will this be a problem if, e.g., I start two builds at the same time (one stagenet, one mainnet)? The fact that the intermediate image is tagged with the same name?

Reply:
yeah it probably could, I added a new param to the build scripts/dockerfile `STATIC_BINARIES_IMAGE_TAG`, in the "manual" builds for test-/mainnet it defaults to latest, and in jenkins it is set to `BUILD_TAG`, which is of the format `jenkins-concordium-node-ubuntu-10`, so that ties it to the specific jenkins build, meaning it should basically never be reused.